make-release.sh

```diff
@@ -0,0 +1,181 @@
+echo "Check out $REPO to ${TMP}/${REPO##*/}"
+gitw clone "$REPO" -q
+cd ./* || exit 1
+
+if [[ "$CREATE_BRANCH" = "true" ]]; then
+  echo "Create branch $BRANCH"
+  if git ls-remote --exit-code --heads origin $BRANCH > /dev/null; then
+    echo "Error: branch $BRANCH already exists in $REPO"
+    exit 1
+  fi
+  gitw checkout -b "$BRANCH"
+  gitw push origin "$BRANCH"
+else
+  gitw fetch origin && gitw checkout "$BRANCH"
+fi
+
+# change VERSION file
+echo "Updating VERSION file to $VERSION"
+echo "$VERSION" > VERSION
+
+# commit change into branch
+COMMIT_MSG="[release] Bump to $VERSION in $BRANCH"
+echo "Creating commit in $BRANCH with message: $COMMIT_MSG"
+gitw pull origin "$BRANCH"
+gitw commit -s -m "$COMMIT_MSG" VERSION
+gitw push origin "$BRANCH"
+
+# force release branch to current commit
+echo "Updating branch 'release' to point to current HEAD"
+gitw branch release -f
+gitw push origin release -f
+
+# tag the release
+echo "Creating and pushing git tag $VERSION"
+gitw checkout "$BRANCH"
+gitw tag "$VERSION"
+gitw push origin "$VERSION"
+
+# now update ${BASE_BRANCH} to the new snapshot version
+echo "Updating SNAPSHOT version in $BASE_BRANCH"
+gitw fetch origin "${BASE_BRANCH}":"${BASE_BRANCH}"
+gitw checkout "${BASE_BRANCH}"
+
+# change VERSION file + commit change into ${BASE_BRANCH} branch
+if [[ "$MAJOR_VERSION_BUMP" = true ]]; then
+  NEXT_VERSION="${NEW_MAJOR_VERSION}.1.0-SNAPSHOT"
+elif [[ "${BASE_BRANCH}" != "${BRANCH}" ]]; then
+  # bump the y digit
+  [[ $BRANCH =~ ^(v[0-9]+)\.([0-9]+)\.x ]] && BASE=${BASH_REMATCH[1]}; NEXT=${BASH_REMATCH[2]}; (( NEXT=NEXT+1 )) # for BRANCH=7.10.x, get BASE=7, NEXT=11
+  NEXT_VERSION="${BASE}.${NEXT}.0-SNAPSHOT"
+else
+  # bump the z digit
+  [[ $VERSION =~ ^(v[0-9]+)\.([0-9]+)\.([0-9]+) ]] && BASE="${BASH_REMATCH[1]}.${BASH_REMATCH[2]}"; NEXT="${BASH_REMATCH[3]}"; (( NEXT=NEXT+1 )) # for VERSION=7.7.1, get BASE=7.7, NEXT=2
+  NEXT_VERSION="${BASE}.${NEXT}-SNAPSHOT"
+fi
+
+# change VERSION file
+echo "Updating version in VERSION to $NEXT_VERSION"
+echo "${NEXT_VERSION}" > VERSION
+
+# commit change into branch
+echo "Committing version change in $BASE_BRANCH"
+COMMIT_MSG="[release] Bump to ${NEXT_VERSION} in ${BASE_BRANCH}"
+gitw pull origin "${BASE_BRANCH}"
+gitw commit -s -m "${COMMIT_MSG}" VERSION
+
+PUSH_TRY="$(gitw push origin "${BASE_BRANCH}")"
+# shellcheck disable=SC2181
+if [[ $? -gt 0 ]] || [[ $PUSH_TRY == *"protected branch hook declined"* ]]; then
+  echo "Could not increment SNAPSHOT version - branch $BASE_BRANCH is protected"
+  PR_BRANCH=pr-${BASE_BRANCH}-to-${NEXT_VERSION}
+  # create pull request for master branch, as branch is restricted
+  gitw checkout -b "${PR_BRANCH}"
+  gitw push origin "${PR_BRANCH}"
+  lastCommitComment="$(git log -1 --pretty=%B)"
+  if command -v hub &> /dev/null; then
+    hub pull-request -o -f -m "${lastCommitComment}
+
+${lastCommitComment}" -b "${BASE_BRANCH}" -h "${PR_BRANCH}"
+  else
+    echo "Could not open PR in $REPO: 'hub' not installed"
```

Comment:
We could check that hub is installed BEFORE getting this far. That way if we're triggering this script from Jenkins, we'll fail faster for a node that doesn't have hub installed. Better that than a half-release (branch created, updated, tagged; master branch not changed).

Reply:
could be something as simple as https://github.com/redhat-developer/codeready-workspaces/blob/master/product/updateBaseImages.sh#L9-L17 but maybe using a `which` check so that install path is more flexible.

Reply:
I don't want to make `hub` a hard requirement for the script though, since I don't want to install hub locally on my machine.